### PR TITLE
Simplify index controller

### PR DESCRIPTION
**Pull request by fobos**
Made small refactoring of index controller:

- got rid of `q` package in case of unnecessity. Just use built-in Promises
- remove `findPrs` from big closure to make controllers code drastically smaller and readable
- simplify pr's merge status fetching by using standard `Promise.all` method
- remove assigning lodash into `app` because we can require it anywhere

### CODE REVIEW

File: controllers/index.js [hunk 1]
```diff
@@ -1,152 +1,118 @@
...
+        return res.render('index');
+    }
 
-            const requests = [];
+    Promise.all([
+        findPrs(github, username),
+        github.users.getForUser({username})
+            .then(logCallsRemaining)
+    ])
+        .then(([prs, user]) => {
```
Comment (jenkoian) [inline controllers/index.js]: Possibly due to the version of node I'm on (I want to target latest LTS version) but this reports the following error for me: 

```
/Users/ian/Code/hacktoberfest-checker/controllers/index.js:36
        .then(([prs, user]) => { 
               ^

SyntaxError: Unexpected token [
    at exports.runInThisContext (vm.js:53:16)
    at Module._compile (module.js:404:25)
    at Object.Module._extensions..js (module.js:432:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:311:12)
    at Module.require (module.js:366:17)
    at require (module.js:385:17)
    at Object.<anonymous> (/Users/ian/Code/hacktoberfest-checker/index.js:16:25)
    at Module._compile (module.js:425:26)
    at Object.Module._extensions..js (module.js:432:10)
error Command failed with exit code 1.
```

**Comment by jenkoian** [comment]
Much nicer, just that one error I think.

File: controllers/index.js (hunk 1 above)
Comment (fobos) [reply controllers/index.js]: What version of node do you use?

Comment (jenkoian) [reply controllers/index.js]: Sorry I just realised I had dipped into version 5 to debug something, works fine on 6+ 👍

**Comment by jenkoian** [comment]
Thanks @fobos !

**Pull request merged**